### ISSUE TITLE
rancher2_app_v2 throws error about inconsistent results with longhorn (wait = true doesn't work)

### ISSUE THREAD

**Issue by andreshamedfernandez** [rancher/terraform-provider-rancher2#600]
When trying to install longhorn using the rancher2_app_v2 I get this error:

```
2021/03/05 10:25:11 [ERROR] eval: *terraform.EvalApplyPost, err: Provider produced inconsistent result after apply: When applying changes to rancher2_app_v2.longhorn-system, provider "registry.terraform.io/rancher/rancher2" produced an unexpected new value: Root resource was present, but now absent.

This is a bug in the provider, which should be reported in the provider's own issue tracker.
[ERROR] eval: *terraform.EvalSequence, err: Provider produced inconsistent result after apply: When applying changes to rancher2_app_v2.longhorn-system, provider "registry.terraform.io/rancher/rancher2" produced an unexpected new value: Root resource was present, but now absent.

This is a bug in the provider, which should be reported in the provider's own issue tracker.

Error: Provider produced inconsistent result after apply

When applying changes to rancher2_app_v2.longhorn-system, provider
"registry.terraform.io/rancher/rancher2" produced an unexpected new value:
Root resource was present, but now absent.

This is a bug in the provider, which should be reported in the provider's own
issue tracker.

```
What I also see in the logging is this warning: 

```
2021/03/05 10:24:50 [WARN] Provider "registry.terraform.io/rancher/rancher2" produced an invalid plan for rancher2_app_v2.longhorn-system, but we are tolerating it because it is using the legacy plugin SDK.
rancher2_app_v2.longhorn-system: Creating...
    The following problems may be the cause of any confusing errors from downstream operations:
      - .cleanup_on_fail: planned value cty.False does not match config value cty.NullVal(cty.Bool)
      - .disable_hooks: planned value cty.False does not match config value cty.NullVal(cty.Bool)
      - .disable_open_api_validation: planned value cty.False does not match config value cty.NullVal(cty.Bool)
      - .force_upgrade: planned value cty.False does not match config value cty.NullVal(cty.Bool)
```

Strange thing is that the longhorn is installed correctly and works fine but the state is saved so everytime we run this it wants to install Longhorn.

This is the configuration I use
```
resource "rancher2_app_v2" "longhorn-system" {
  cluster_id = var.rancher_cluster_id
  name = "Longhorn"
  namespace = "longhorn-system"
  repo_name = "rancher-charts"
  chart_name = "longhorn"
  chart_version = "1.1.000"
  values = file("${path.module}/values/values.yaml")
  wait = true
}
```

Version being use are as follows:

- Using hashicorp/kubernetes v1.13.3 from the shared cache directory
- Using rancher/rancher2 v1.10.6 from the shared cache directory
- Using Terraform v0.13.2
- Using Terragrunt version v0.23.40 

Thanks in advance for the help

**Comment by rawmind0** [comment]
This issue seems related to a race condition, where trying to read app v2 before it's available at Rancher API. Added retry logic to get V2 catalogs and apps at PR #569 . This should address the issue.

> What I also see in the logging is this warning:

These warning shouldn't be related to the issue, they are related to legacy sdk used by this provider, but just warnings. We are working to update the sdk used by the provider.

**Comment by rawmind0** [comment]
Fix included on new provider release v1.12.0

Please, reopen issue if needed
